server/app/socketService.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable consistent-return */
+const SocketIo = require('socket.io');
+const SocketClient = require('socket.io-client');
+const httpServer = require('http').createServer();
+const User = require('./models/User');
+
+const ENDPOINT = process.env.SOCKET_CHANNEL_ENDPOINT || 'http://localhost';
+const PORT = process.env.SOCKET_CHANNEL_PORT || 3002;
+
+const io = SocketIo();
+const socketServer = {};
+
+/*
+ *  List of connected clients
+ *  {
+ *    [userId]: SocketObject
+ *  }
+ */
+socketServer.clients = {};
+
+socketServer.start = server => {
+  io.attach(server);
+  io.on('connection', socket => {
+    if (process.env.NODE_ENV !== 'production') {
+      console.log(`Socket: new connection from ${socket.id}`);
+    }
+
+    setTimeout(() => {
+      if (!socket.authenticated) return socket.disconnect(true);
+    }, 2000);
+
+    // Use user email to determin the identity of client
+    socket.on('authenticate', ({ email }) => {
+      if (!email) return socket.disconnect(true);
+
+      User.findOne({ email }, (err, user) => {
+        if (err) {
+          console.log(err);
+          return socket.disconnect(true);
+        }
+
+        if (!user) return socket.disconnect(true);
+
+        // eslint-disable-next-line no-param-reassign
+        socket.authenticated = true;
+
+        socket.on('disconnect', () => {
+          delete socketServer.clients[user._id];
+        });
+
+        // Add current connection to list of connected clients
+        socketServer.clients[user._id] = socket;
+
+        if (process.env.NODE_ENV !== 'production') {
+          console.log(Object.keys(socketServer.clients));
+        }
+      });
+    });
+  });
+
+  // A separate endpoint for triggering the emittance of 'notification' event
+  const messageChannel = SocketIo(
+    httpServer.listen(PORT, err => {
```

Comment:
seems like this isn't actually used, because you attach the other server

Reply:
This server is for receiving messages from the web scraping service